checks.d/openstack.py

```diff
@@ -0,0 +1,827 @@
+            self.service_check(self.IDENTITY_API_SC, AgentCheck.CRITICAL, tags=["server:%s" % self.init_config.get("keystone_server_url")])
+
+        except IncompleteConfig as e:
+            if isinstance(e, IncompleteAuthScope):
+                self.warning("""Please specify the auth scope via the `auth_scope` variable in your init_config.\n
+                             The auth_scope should look like: \n
+                            {'project': {'name': 'my_project', 'domain': {'id': 'my_domain_id'}}}\n
+                            OR\n
+                            {'project': {'id': 'my_project_id'}}
+                             """)
+            elif isinstance(e, IncompleteIdentity):
+                self.warning("Please specify the user via the `user` variable in your init_config.\n" +
+                             "This is the user you would use to authenticate with Keystone v3 via password auth.\n" +
+                             "The user should look like: {'password': 'my_password', 'name': 'my_name', 'domain': {'id': 'my_domain_id'}}")
+            else:
+                self.warning("Configuration Incomplete! Check your openstack.yaml file")
+
+        else:
+            # All is good, send the word up to DD
+            self.service_check(self.NETWORK_API_SC, AgentCheck.OK, tags=["keystone_server:%s" % self.init_config.get("keystone_server_url")])
+            self.service_check(self.COMPUTE_API_SC, AgentCheck.OK, tags=["keystone_server:%s" % self.init_config.get("keystone_server_url")])
+            self.service_check(self.IDENTITY_API_SC, AgentCheck.OK, tags=["server:%s" % self.init_config.get("keystone_server_url")])
+
+
+    #### Local Info accessors
+    def get_local_hypervisor(self):
+        """
+        Returns the hypervisor running on this host, and assumes a 1-1 between host and hypervisor
+        """
+        # Look up hypervisors available filtered by my hostname
+        host = self.get_my_hostname()
+        hyp = self.get_all_hypervisor_ids(filter_by_host=host)
+        if hyp:
+            return hyp[0]
+
+    def get_scoped_project(self, instance):
+        """
+        Returns the project that this instance of the check is scoped to
+        """
+        project_auth_scope = self.get_scope_for_instance(instance)
+        if project_auth_scope.tenant_id:
+            return {"id": project_auth_scope.tenant_id}
+
+        filter_params = {
+            "name": project_auth_scope.project_name,
+            "domain_id": project_auth_scope.domain_id
+        }
+
+        url = "{0}/{1}/{2}".format(self.keystone_server_url, DEFAULT_KEYSTONE_API_VERSION, "projects")
+        headers = {'X-Auth-Token': self.get_auth_token(instance)}
+
+        projects = []
+        try:
+            resp = self._make_request_with_auth_fallback(url, headers, params=filter_params)
+            project_details = resp.json()
```

Comment:
Looks like you're always calling `.json` everytime you call `_make_request_with_auth_fallback` so it should likely be part of the method.